writer.go

```diff
@@ -0,0 +1,80 @@
+package kinesis
+
+import (
+	"time"
+
+	"github.com/gliderlabs/logspout/router"
+)
+
+type writer struct {
+	buffer         *buffer
+	flusher        Flusher
+	messages       chan *router.Message
+	buffers        chan buffer
+	dropBufferFunc func(buffer)
+	ticker         <-chan time.Time
+}
+
+func newWriter(b *buffer, f Flusher) *writer {
+	w := &writer{
+		messages:       make(chan *router.Message),
+		buffers:        make(chan buffer, 10),
+		ticker:         time.NewTicker(time.Second).C,
+		dropBufferFunc: dropBuffer,
+		flusher:        f,
+		buffer:         b,
+	}
+
+	return w
+}
+
+// Start starts the goroutines consuming, dropping and flushing messages.
+func (w *writer) Start() {
+	go w.bufferMessages()
+	go w.flushBuffers()
+}
+
+func (w *writer) write(m *router.Message) {
+	w.messages <- m
+}
+
+func (w *writer) bufferMessages() {
+	flush := func() {
+		select {
+		case w.buffers <- *w.buffer:
```

Comment:
`flusher` only ever accesses `buffer.input`, so it should probably take a `*kinesis.PutRecordsInput` instead.

I think we should also move all of the logic for dropping into the `flusher` implementation, which makes writer a lot simpler since you can just do:

```go
if w.buffer.full(m) {
        w.flusher.flush(w.buffer.input)
}
```